### PR TITLE
Better system messages if you did something

### LINKED ISSUE #1145: Better system messages if you did something to yourself
E.g. if I remove myself from the conversation, it will say "@mario removed @Mario from the conversation".

Instead it should say: "Mario left the conversation".

### PR DESCRIPTION
**Pull request by nickvergessen**
Fix #1145

### CODE REVIEW

File: lib/Chat/SystemMessage/Parser.php [hunk 1]
```diff
@@ -55,40 +59,93 @@ public function parseMessage(IComment $comment, string $displayName): array {
...
+					$parsedMessage = $this->l->t('{actor} removed you from the conversation');
+				}
+			}
 		} else if ($message === 'moderator_promoted') {
 			$parsedParameters['user'] = $this->getUser($parameters['user']);
 			$parsedMessage = $this->l->t('{actor} promoted {user} to moderator');
+			if ($currentUserIsActor) {
+				$parsedMessage = $this->l->t('You promoted {user} to moderator');
+			} else if ($currentUser instanceof IUser && $currentUser->getUID() === $parsedParameters['user']['id']) {
+				$parsedMessage = $this->l->t('{actor} promoted you from moderator');
```
Comment (Ivansss) [inline lib/Chat/SystemMessage/Parser.php]: This should be "promoted you to moderator"

Comment (mario) [reply lib/Chat/SystemMessage/Parser.php]: @Ivansss is right.

File: lib/Chat/SystemMessage/Parser.php [hunk 2]
```diff
@@ -55,40 +59,93 @@ public function parseMessage(IComment $comment, string $displayName): array {
...
+			}
 		} else if ($message === 'moderator_demoted') {
 			$parsedParameters['user'] = $this->getUser($parameters['user']);
 			$parsedMessage = $this->l->t('{actor} demoted {user} from moderator');
+			if ($currentUserIsActor) {
+				$parsedMessage = $this->l->t('You demoted {user} from moderator');
+			} else if ($currentUser instanceof IUser && $currentUser->getUID() === $parsedParameters['user']['id']) {
+				$parsedMessage = $this->l->t('{actor} demoted you from moderator');
```
Comment (mario) [inline lib/Chat/SystemMessage/Parser.php]: This needs better wording. Demoted from moderator is ... bad.

Comment (nickvergessen) [reply lib/Chat/SystemMessage/Parser.php]: that is the proper englisch term

File: lib/Chat/SystemMessage/Parser.php (hunk 1 above)
Comment (nickvergessen) [reply lib/Chat/SystemMessage/Parser.php]: fixed

**Pull request merged**